GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace GenericsHomework;
+
+public class Node<TVal> where TVal : notnull
+{
+    public Node(TVal val)
```

Comment:
```suggestion
    public Node(TVal value)
```
Consider using non-abbreivated names